assets/dev/js/frontend/handlers/section.js

```diff
@@ -1,3 +1,155 @@
+class BackgroundSlideshow extends elementorModules.frontend.handlers.Base {
+	getDefaultSettings() {
+		return {
+			selectors: {
+				swiperContainer: '.elementor-section-background-slideshow',
+				swiperSlides: '.swiper-slide',
+			},
+			classes: {
+				swiperContainer: 'elementor-section-background-slideshow swiper-container',
+				swiperWrapper: 'swiper-wrapper',
+				swiperSlide: 'elementor-section-background-slideshow__slide swiper-slide',
+				swiperSlideInner: 'elementor-section-background-slideshow__slide__image',
+			},
+		};
+	}
+
+	getSwiperOptions() {
+		const elementSettings = this.getElementSettings();
+
+		const swiperOptions = {
+			grabCursor: false,
+			slidesPerView: 1,
+			slidesPerGroup: 1,
+			loop: 'yes' === elementSettings.background_slideshow_loop,
+			speed: elementSettings.background_slideshow_animation_speed,
+			autoplay: {
+				delay: elementSettings.background_slideshow_autoplay_speed,
+				disableOnInteraction: false,
+				stopOnLastSlide: 'yes' !== elementSettings.background_slideshow_loop,
+			},
+		};
+
+		if ( 'yes' === elementSettings.background_slideshow_loop ) {
+			swiperOptions.loopedSlides = this.getSlidesCount();
+		}
+
+		switch ( elementSettings.background_slideshow_content_animation ) {
+			case 'fade':
+				swiperOptions.effect = 'fade';
+				break;
+			case 'slide_right':
+				swiperOptions.initialSlide = this.getSlidesCount() - 1;
+				swiperOptions.autoplay.reverseDirection = true;
+				break;
+			case 'slide_left':
+				break;
+			case 'slide_up':
+				swiperOptions.direction = 'vertical';
+				break;
+			case 'slide_down':
+				swiperOptions.direction = 'vertical';
+				swiperOptions.autoplay.reverseDirection = true;
+				break;
+		}
+
+		return swiperOptions;
+	}
+
+	getSlidesCount() {
+		return this.elements.$slides.length;
+	}
+
+	buildSwiperElements() {
+		const classes = this.getSettings( 'classes' ),
+			elementSettings = this.getElementSettings();
+
+		const $container = jQuery( '<div>', { class: classes.swiperContainer } ),
+			$wrapper = jQuery( '<div>', { class: classes.swiperWrapper } );
+
+		elementSettings.background_slideshow_gallery.forEach( ( slide ) => {
+			const $slide = jQuery( '<div>', { class: classes.swiperSlide } ),
+				$slidebg = jQuery( '<div>', {
+				class: classes.swiperSlideInner,
+				style: 'background-image: url("' + slide.url + '");',
+			} );
+
+			$slide.append( $slidebg );
+			$wrapper.append( $slide );
+		} );
+
+		$container.append( $wrapper );
+
+		this.$element.prepend( $container );
+	}
+
+	getSlideshowElements() {
+		const selectors = this.getSettings( 'selectors' );
+
+		this.elements = {
+			$backgroundSlideShowContainer: this.$element.find( selectors.swiperContainer ),
+		};
+
+		this.elements.$slides = this.elements.$backgroundSlideShowContainer.find( selectors.swiperSlides );
+	}
+
+	initSlider() {
+		this.getSlideshowElements();
+
+		const $slider = this.elements.$backgroundSlideShowContainer;
+
+		if ( ! $slider.length ) {
```

Comment:
Redundant check

Reply:
why? 
If there is any problem preventing buildSwiperElements() from adding the HTML markup, there will be no HTML element to initialize Swiper on. If Swiper is initialized with a non-existant element, it throws an error.